src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueue.java

```diff
@@ -135,7 +135,12 @@
     private int cycleFileRenamed = -1;
     @NotNull
     private Condition createAppenderCondition = NoOpCondition.INSTANCE;
-    protected final ThreadLocal<ExcerptAppender> strongExcerptAppenderThreadLocal = CleaningThreadLocal.withCloseQuietly(this::createNewAppenderOnceConditionIsMet);
+    protected final ThreadLocal<ExcerptAppender> strongExcerptAppenderThreadLocal = CleaningThreadLocal.withCloseQuietly(() -> {
+        final StoreAppender newAppenderOnceConditionIsMet = (StoreAppender) this.createNewAppenderOnceConditionIsMet();
+        // Thread-local Appenders should prevent single-threaded checks being disabled/reset
+        newAppenderOnceConditionIsMet.bindToCreatingThread();
```

Comment:
This needs a corresponding change in core to ignore UnsupportedOperationException when `net.openhft.chronicle.core.io.AbstractCloseable#waitForCloseablesToClose` calls `singleThreadCheckDisabled` on the appender.

Reply:
Actually had to disable this because `PublishHandler` (some part of channels) attempts to disable thread safety checks on Appenders for some reason.

I would like to put it in though.